packages/ses/README.md

```diff
@@ -74,6 +74,47 @@ c1.global === c2.global; // false
 c1.global.JSON === c2.global.JSON; // true
 ```
 
+### Compartments
+
+Any code executed within a compartment shares a set of module instances.
+For modules to work within a compartment, the creator must provide
+a `resolveHook` and an `importHook`.
+The `resolveHook` determines how the compartment will infer the full module
+specifier for another module from a referrer module and the module specifier
+imported within that module.
+The `importHook` accepts a module specifier and asynchronously returns a
+`ModuleStaticRecord` for that module.
+
+```js
+import { Compartment, ModuleStaticRecord } from 'ses';
+const c1 = new Compartment({}, {}, {
+  resolveHook: (moduleSpecifier, moduleReferrer) => {
+    return new URL(moduleSpecifier, moduleReferrer).toString();
+  },
+  importHook: async moduleSpecifier => {
+    const moduleLocation = locate(moduleSpecifier);
+    const moduleText = await retrieve(moduleLocation);
+    return new ModuleStaticRecord(moduleText);
+  },
+});
+```
+
+A compartment can also link a module in another compartment.
+Each compartment has a `module` function that accepts a module specifier
+and returns the ModuleNamespace for that module.
+The ModuleNamespace is not useful for inspecting the exports of the
+module until that module has been imported, but it can be passed into the
+module map of another Compartment, creating a link.
+
+```js
+const c2 = new Compartment({}, {
+  'https://example.com/packages/example/': c1.module('./main.js'),
```

Comment:
This example amplifies confusion between full specifiers and moduleLocations. The keys in a moduleMap are full specifiers, which will typically not be absolute urls. moduleLocations will typically be absolute urls.

Reply:
Indeed. The tension is that, in the example, I can use the `URL` to make a trivial and obvious `resolveHook` without either including a bunch more code or invoking imaginary code. I believe I’ll have to invoke imaginary code for now, and fix this when the imaginary code is in a public package, named something like `@agoric/ses-plugin-node-resolver`.

Reply:
(Though, this example is certainly legitimate usage and illustrates how to emulate the web’s rendition of import.) I agree that there’s a danger of confusion here. Plan of record is to invoke an imaginary function here.